### DISCUSSION TITLE
Why is the cachefile zpool property not persisted?

### DISCUSSION THREAD

**Discussion by FiloSottile** [openzfs/zfs#11757] [Q&A]
I have a pool with a custom `cachefile`, let's say `/var/tmp/zpool.cache`.

After a reboot, I run

```
zpool import -c /var/tmp/zpool.cache -a
```

The import succeeds, but the `cachefile` setting is back to the default.

```
NAME  PROPERTY   VALUE                         SOURCE
test  cachefile  -                             default
```

I would have expected that either

1. the zpool property would be persisted in the pool, like all others
2. the `-c` argument would be used as the `cachefile` of the imported pool

What am I missing?

**Comment by ghost** [comment]
With `zpool import -c` the given cachefile is only used to find the pools for importing. It does not change the property on the pools. To do that you should use `zpool set`.

**Reply by FiloSottile** [reply]
That's fair enough on `-c`, although I wonder what users want to use a cache file for importing, but then not to track changes to the pool.

However, I _did_ also use `zpool set` (that's how the cache file was created, I did not move it manually) and the value was back to the default after `zpool import`.

It's deeply surprising that I explicitly set a custom cache file, update my boot script to pass it to `zpool import -c`, and then at the next boot the cache file gets recreated at `/etc/zfs/zpool.cache` anyway.

**Reply by ghost** [reply]
Ok I understand your concern now, that is indeed surprising.
